aboutNewTabService.js

```diff
@@ -0,0 +1,252 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+"use strict";
+
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/AppConstants.jsm");
+
+ChromeUtils.defineModuleGetter(this, "AboutNewTab",
+                               "resource:///modules/AboutNewTab.jsm");
+
+const TOPIC_APP_QUIT = "quit-application-granted";
+const TOPIC_LOCALES_CHANGE = "intl:app-locales-changed";
```

Comment:
This file seems to be outdated? In particular missing some about:welcome and @imjching TOPIC_CONTENT_DOCUMENT_INTERACTIVE stuff

Reply:
i need to stop using dxr and make the switch to searchfox. will update

Reply:
or maybe i'll just pull latest mc and it from there